src/factory/draft/preset_discovery.rs

```diff
@@ -0,0 +1,228 @@
+            os_error: i32,
+            error_message: *const c_char,
+        ),
+    >,
+    pub begin_preset: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            name: *const c_char,
+            load_key: *const c_char,
+        ) -> bool,
+    >,
+    pub add_plugin_id: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            plugin_id: *const clap_plugin_id,
+        ),
+    >,
+    pub set_soundpack_id: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            soundpack_id: *const c_char,
+        ),
+    >,
+    pub set_flags: Option<
+        unsafe extern "C" fn(receiver: *const clap_preset_discovery_metadata_receiver, flags: u32),
+    >,
+    pub add_creator: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            creator: *const c_char,
+        ),
+    >,
+    pub set_description: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            description: *const c_char,
+        ),
+    >,
+    pub set_timestamps: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            creation_time: clap_timestamp,
+            modification_time: clap_timestamp,
+        ),
+    >,
+    pub add_feature: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            feature: *const c_char,
+        ),
+    >,
+    pub add_extra_info: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            key: *const c_char,
+            value: *const c_char,
+        ),
+    >,
+}
+
+unsafe impl Send for clap_preset_discovery_metadata_receiver {}
+unsafe impl Sync for clap_preset_discovery_metadata_receiver {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_filetype {
+    pub name: *const c_char,
+    pub description: *const c_char,
+    pub file_extension: *const c_char,
+}
+
+unsafe impl Send for clap_preset_discovery_filetype {}
+unsafe impl Sync for clap_preset_discovery_filetype {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_location {
+    pub flags: u32,
+    pub name: *const c_char,
+    pub uri: *const c_char,
+}
+
+unsafe impl Send for clap_preset_discovery_location {}
+unsafe impl Sync for clap_preset_discovery_location {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_soundpack {
+    pub flags: u32,
+    pub id: *const c_char,
+    pub name: *const c_char,
+    pub description: *const c_char,
+    pub homepage_url: *const c_char,
+    pub vendor: *const c_char,
+    pub image_url: *const c_char,
```

Comment:
Should be `image_uri`: https://github.com/free-audio/clap/blob/065d685d4e9657f0344f350eef748be2b4d8e318/include/clap/factory/draft/preset-discovery.h#L209